frontend/src/certificate_verification.tsx

```diff
@@ -0,0 +1,266 @@
+
+import { Individual } from "@app/data_context_global";
+import { styles } from "@material-ui/pickers/views/Calendar/Calendar";
+import { useMessageContext } from "@app/message_context";
+
+const useStyles = makeStyles({
+  titleText: {
+    width: "100%",
+    borderBottom: "1px solid lightgrey",
+    padding: "0 20px",
+    fontSize: "2.3em",
+  },
+  fileInput: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  label: {
+    margin: "20px 20px 20px 0",
+    marginRight: "20px",
+  },
+  flexRow: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "flex-end",
+    margin: "1em 0",
+  },
+  responseBox: {
+    width: "100%",
+    padding: "1em",
+    margin: "2em 0",
+  },
+  successIcon: {
+    fill: "#388e3c", // same as success.dark in the default theme
+    marginLeft: "0.5em",
+  },
+  warnIcon: {
+    fill: "#ff9800", // same as warning.main in the default theme
+    marginLeft: "0.5em",
+  },
+  failIcon: {
+    fill: "#d32f2f", // same value as error.dark in the default theme
+    marginLeft: "0.5em",
+  },
+  boxTitle: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+  },
+});
+
+export function CertificateVerification({
+  id,
+  individual,
+}: {
+  id: string;
+  individual: Individual | undefined;
+}) {
+  const [file, setFile] = React.useState(undefined as File | undefined);
+  const [certValid, setCertValid] = React.useState(false as boolean);
+  const [certOutdated, setCertOutdated] = React.useState(false as boolean);
+  const [certNotFound, setCertNotFound] = React.useState(false as boolean);
+  const [showPdf, setShowPdf] = React.useState(false as boolean);
+  //States to make pdf-preview-library work
+  const [numPages, setNumPages] = React.useState(null);
+  const [pageNumber, setPageNumber] = React.useState(1);
+  const [previewUrl, setPreviewUrl] = React.useState(undefined as unknown);
+  const [previewWidth, setPreviewWidth] = React.useState(0 as Number);
+  const style = useStyles();
+  const { userMessage } = useMessageContext();
+
+  const verifyCertificate = (id: string) => {
+    const postCertificate = (
+      id: string,
+      content: string | ArrayBuffer | undefined
+    ) => {
+      fetch(`/api/certificates/verify/${id}`, {
+        body: content,
+        method: "POST",
+        credentials: "same-origin",
+        headers: {
+          "Content-Type": "application/pdf",
+        },
+      })
+        .then((res) => {
+          if (res.status === 200) {
+            setCertValid(true);
+          } else if (res.status === 202) {
+            setCertOutdated(true);
+          } else if (res.status === 404) {
+            setCertNotFound(true);
+          } else {
+            throw new Error("Något gick fel.");
+          }
+        })
+        .catch((error) => {
+          userMessage(error.message, "error");
+        });
+    };
+
+    if (file) {
+      const reader = new FileReader();
```

Comment:
I can't see it being an issue here, but if I understand this correctly (which might not be the case), it will first load the file and then post it. If possible, streaming it would be preferential.

(E.g. what happens if I pick a 50 Gbyte file rather than the one I intended.)

Reply:
(I think API support used to be iffy, but I have not kept up to date and wouldn't know about anything resembling modern things.)